server/controllers/placeController.js

```diff
@@ -0,0 +1,112 @@
+import PlaceHelpers from '../helpers/placeHelper';
+import pagination from '../helpers/paginationHelper';
+
+/**
+ * This class contains all methods
+ * required to handle
+ * trip endpoints' request.
+ */
+class PlaceController {
+  /**
+   * This method handle createPlace request.
+   * @param {object} req The user's request.
+   * @param {object} res The response.
+   * @returns {object} The status and some data of the place.
+   */
+  static async createPlace(req, res) {
+    try {
+      const { body } = req;
+      const newPlace = {
+        name: body.name,
+        country: body.country,
+        city: body.city,
+      };
+      const foundPlace = await PlaceHelpers.placeExist('name', body.name);
+      if (foundPlace.length !== 0) {
+        return res.status(409).json({
+          status: 409,
+          data: 'This place already exists, use diferent name.'
+        });
+      }
+      const savePlace = await PlaceHelpers.savePlace(newPlace);
+
+      return res.status(201).json({
+        status: 201,
+        message: 'place was created successfully.',
+        data: savePlace
+      });
+    } catch (error) {
+      return res.status(500).json({
+        status: 500,
+        message: 'Something went wrong',
+        error: error.message
+      });
+    }
+  }
+
+  /**
+   * This method handles view all trip requests.
+   * @param {object} req The user's request.
+   * @param {object} res The response.
+   * @param {function} next The next action.
+   * @returns {object} The status and some data of the trip.
+   */
+  static async retrieveAllPlaces(req, res, next) {
+    try {
+      const {
+        start, end, pages, skip, paginate
+      } = await pagination.paginateData(req.query);
+      const { rows, count } = await PlaceHelpers.allPlace(skip, start);
+
+      const userAllData = rows;
```

Comment:
These are explicitly places not user data. So, use a corresponding variable.

Reply:
This variable are the variable pagination function expect if i change them they will break and pagination will not be reusable anymore.

Reply:
If so, find a general variable name that will fir each case, instead of reusing the one that refers to paginating retrieved users.

Reply:
Feedback implemented